### PR TITLE
[MOBILE-1650] Add badge methods for iOS

### PR DESCRIPTION
**Pull request by rlepinski**
### What do these changes do?
Adds badge methods and properties for iOS.

### How did you verify these changes?
Ran sample, verified setters/getters.

### CODE REVIEW

File: ios/Classes/TiAirshipModule.m [hunk 1]
```diff
@@ -118,6 +118,26 @@ - (void)setUserNotificationsEnabled:(id)value {
     [[UAirship push] updateRegistration];
 }
 
+- (id)isAutoBadgeEnabled {
+    return NUMBOOL([UAirship push].autobadgeEnabled);
+}
+
+- (void)setIsAutoBadgeEnabled:(id)args {
+    [UAirship push].autobadgeEnabled = [TiUtils boolValue:args def:YES];
```
Comment (BrianBatchelder) [inline ios/Classes/TiAirshipModule.m]: Just wondering if the default should be the current value, i.e. a no-op.

File: ios/Classes/TiAirshipModule.m [hunk 2]
```diff
@@ -118,6 +118,26 @@ - (void)setUserNotificationsEnabled:(id)value {
...
+}
+
+- (id)badgeNumber {
+    return NUMINTEGER([UAirship push].badgeNumber);
+}
+
+- (void)setBadgeNumber:(id)args {
+    [UAirship push].badgeNumber = [TiUtils boolValue:args def:YES];
```
Comment (BrianBatchelder) [inline ios/Classes/TiAirshipModule.m]: Shouldn't this be an int rather than a bool?

**Review by BrianBatchelder — CHANGES_REQUESTED** [review]
Couple things.

File: ios/Classes/TiAirshipModule.m (hunk 2 above)
Comment (rlepinski) [reply ios/Classes/TiAirshipModule.m]: yeah, i was testing with 1 so i didnt notice...

**Review by BrianBatchelder — APPROVED** [review]
:thumbsup:

File: ios/Classes/TiAirshipModule.m [hunk 3]
```diff
@@ -118,6 +118,28 @@ - (void)setUserNotificationsEnabled:(id)value {
     [[UAirship push] updateRegistration];
 }
 
+- (id)isAutoBadgeEnabled {
+    return NUMBOOL([UAirship push].autobadgeEnabled);
+}
+
+- (void)setIsAutoBadgeEnabled:(id)args {
+    [UAirship push].autobadgeEnabled = [TiUtils boolValue:args
+                                                      def:[UAirship push].autobadgeEnabled];
```
Comment (marc-scig) [inline ios/Classes/TiAirshipModule.m]: what are the conditions in which args could be nil? I'm wondering if maybe we want to do a similar thing in setUserNotificationsEnabled, which currently has YES as the default value.

File: ios/Classes/TiAirshipModule.m [hunk 4]
```diff
@@ -118,6 +118,28 @@ - (void)setUserNotificationsEnabled:(id)value {
...
+    [[UAirship push] updateRegistration];
+}
+
+- (id)badgeNumber {
+    return NUMINTEGER([UAirship push].badgeNumber);
+}
+
+- (void)setBadgeNumber:(id)args {
+    [UAirship push].badgeNumber = [TiUtils numberFromObject:args].integerValue;
```
Comment (marc-scig) [inline ios/Classes/TiAirshipModule.m]: Does this need a default value? Or likewise could the other one be done with:

`[TiUtils numberFromObject:args].boolValue`

**Comment by rlepinski** [comment]
Cleaned up arg handling and fixed the custom event method to accept an object instead of being forced to jsonify. I left handling both to keep backwards compatibility.

File: ios/Classes/TiAirshipModule.m [hunk 5]
```diff
@@ -165,19 +187,33 @@ - (void)associateIdentifier:(id)args {
...
+        }
+        NSError *jsonError;
+        NSData *data = [customEventString dataUsingEncoding:NSUTF8StringEncoding];
+        payload = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&jsonError];
+        if (jsonError) {
+            UA_LERR(@"Invalid custom event payload %@", customEventString);
+            return;
+        }
+    } else {
+        ENSURE_SINGLE_ARG(args, NSDictionary);
```
Comment (marc-scig) [inline ios/Classes/TiAirshipModule.m:208]: I'm a little confused by this, should it be `ENSURE_SINGLE_ARG(args[0], NSDictionary)`?  How can args have an index and be a dictionary at the same time?

Comment (rlepinski) [reply ios/Classes/TiAirshipModule.m:208]: Magic. The macro redefines the variable

**Review by marc-scig — APPROVED** [review]

**Pull request merged**